src/hazelcore/UHExp.re

```diff
@@ -308,10 +308,130 @@ let text_operand =
   | InvalidTextShape(t) => new_InvalidText(u_gen, t)
   };
 
-let associate = (seq: seq) => {
-  let skel_str = Skel.mk_skel_str(seq, Operators_Exp.to_parse_string);
-  let lexbuf = Lexing.from_string(skel_str);
-  SkelExprParser.skel_expr(SkelExprLexer.read, lexbuf);
+let associate = (seq: seq): skel => {
```

Comment:
Let's also migrate `UHPat` and `UHTyp` to use shunting yard too. I'd move the core logic of `associate` to `Skel` (maybe just call it `mk` in that module), add some extra args that allow clients to specify precedence and whatnot, then instantiate for each `UHExp`, `UHPat`, `UHTyp` module.

Once you've done that, you can delete the `Skel[Expr|Pat|Typ][Parser|Lexer]` files.

Reply:
Will look into all of this. One comment: I wrote a function to look up precedence/associativity of operators in Operators_Exp. Do you still want arguments like this or is that sufficient?

Reply:
There will be a single higher-order `Skel.mk` function that takes a function `precedence: 'operator => int` as an argument, then each `UHExp`/`UHPat`/`UHTyp` module will pass in `Operators_[Exp|Pat|Typ].precedence` function into `Skel.mk`.

Reply:
Ok makes sense.